src/isar_turtlebot/ros_bridge/topic.py

```diff
@@ -68,3 +73,84 @@ def on_message(self, message: dict) -> None:
 
     def subscribe(self) -> None:
         self.topic.subscribe(self.on_message)
+
+
+class ImageTopic(TopicInterface):
+    def __init__(
+        self,
+        client: Ros,
+        name: str,
+        message_type: str,
+        throttle_rate: int = 1000,
+        storage_folder: Path = Path(config.get("storage", "storage_folder")),
+        queue_size: int = 100,
+        queue_length: int = 0,
+        log_callbacks: bool = False,
+    ) -> None:
+        self.name: str = name
+        self.topic: RosTopic = RosTopic(
+            ros=client,
+            name=name,
+            message_type=message_type,
+            throttle_rate=throttle_rate,
+            queue_size=queue_size,
+            queue_length=queue_length,
+        )
+
+        self.log_callbacks: bool = log_callbacks
+        if self.log_callbacks:
+            self.logger: Logger = logging.getLogger("turtlebot_bridge")
+
+        self.value: Optional[Any] = None
+        self.storage_folder: Path = storage_folder
+        self.filenames: dict = dict()
+        self.current_filename: Optional[Path] = None
+
+        self.should_capture_image: bool = False
+
+        self.subscribe()
+
+    def publish(self, message: Any) -> None:
+        self.topic.publish(Message(message))
+
+    def get_value(self) -> Optional[Any]:
+        return self.value
+
+    def on_image(self, message: dict) -> None:
+        image_data = message["data"].encode("ascii")
+        image_bytes = base64.b64decode(image_data)
+
+        self.value = image_bytes
```

Comment:
Value is very generic considering it is used to store image-bytes

Reply:
Will remove `self.value` and getter-function, as this is not in use in the `ImageTopic` class.